fastlane_core/lib/fastlane_core/build_watcher.rb

```diff
@@ -0,0 +1,78 @@
+module FastlaneCore
+  class BuildWatcher
+    def self.find_build(app = nil)
+      build = nil
+      app.latest_version.candidate_builds.each do |b|
+        if !build or b.upload_date > build.upload_date
+          build = b
+        end
+      end
+      unless build
+        UI.user_error!("No processing builds available for app #{app.bundle_id}")
+      end
+      return build
+    end
+
+    def self.wait_for_train(app = nil, platform = "ios", sleep_time = 30)
+      loop do
+        sleep(sleep_time)
+        if app.build_trains(platform: platform).count == 0
+          UI.message("New application; waiting for build train to appear on iTunes Connect")
+        else
+          break
+        end
+      end
+    end
+
+    def self.wait_for_build(app = nil, platform = "ios", sleep_time = 0)
+      start_time = Time.now
+      UI.user_error!("Could not find app with app identifier #{app.bundle_id}") unless app
+      # if this is a new version/app wait for train to show up.
+      if app.build_trains(platform: platform).count == 0
+        self.class.wait_for_train(app, platform, sleep_time)
+      end
+
+      latest_build = nil
+      loop do
+        if sleep_time > 0
+          sleep(sleep_time)
+        else
+          return nil
+        end
+        builds = app.all_processing_builds(platform: platform)
+        break if builds.count == 0
```

Comment:
hey @negebauer - i am unsure how to apply your pr, could you help me with this?

best would be if you checkout this branch, and modify it push it to your fork, so i can have a look.


to test run a PR/branch from a user see this here: https://gist.github.com/hjanuschka/3040fc6f371f46acff72a5e67f649b27